corehq/apps/app_manager/suite_xml/post_process/instances.py

```diff
@@ -186,7 +186,7 @@ def related_locations_fixture_instances(domain, instance_name):
 
 
 def get_all_instances_referenced_in_xpaths(domain, xpaths):
-    instance_re = r"""instance\(['"]([\w\-:]+)['"]\)"""
+    instance_re = br"""instance\(['"]([\w\-:]+)['"]\)"""
```

Comment:
I'm assuming this is necessary because `xpaths` is a list of bytes objects? I think that would mean `instance_names` extracted with this regexp will also be a list of bytes objects, and therefore line 196 will most likely fail because the split string (`':'`) is text (Unicode).

https://github.com/dimagi/commcare-hq/blob/2322dfbe8d21b5366f314749b8b7affb7b75c2cf/corehq/apps/app_manager/suite_xml/post_process/instances.py#L196

Did my reasoning go wrong somewhere?

If not, seems like maybe a better fix might be to decode the items in `xpaths` and do all text processing in this method with text instead of bytes.

Reply:
Good point.  I reverted this change and will address this function in a later PR.